bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java

```diff
@@ -669,15 +716,36 @@ private void validateNonPrimitiveArray(TomlNode tomlValue, String variableName,
     }
 
     private void validateMapValueArray(TomlNode tomlValue, String variableName, ArrayType arrayType, Type elementType) {
-        if (tomlValue.kind() != TomlType.TABLE_ARRAY) {
-            invalidTomlLines.add(tomlValue.location().lineRange());
-            throw new ConfigException(CONFIG_INCOMPATIBLE_TYPE, getLineRange(tomlValue), variableName, arrayType,
-                    getTomlTypeString(tomlValue));
+        TomlType tomlType = tomlValue.kind();
+        if (tomlType == TomlType.TABLE_ARRAY) {
+            visitedNodes.add(tomlValue);
+            List<TomlTableNode> tableNodeList = ((TomlTableArrayNode) tomlValue).children();
+            validateArraySize(tomlValue, variableName, arrayType, tableNodeList.size());
+            for (TomlTableNode tomlTableNode : tableNodeList) {
+                validateValue(tomlTableNode, variableName, elementType);
+            }
+            return;
         }
-        visitedNodes.add(tomlValue);
-        List<TomlTableNode> tableNodeList = ((TomlTableArrayNode) tomlValue).children();
-        for (TomlTableNode tomlTableNode : tableNodeList) {
-            validateValue(tomlTableNode, variableName, elementType);
+        if (tomlType == TomlType.ARRAY) {
+            visitedNodes.add(tomlValue);
+            List<TomlValueNode> tableNodeList = ((TomlArrayValueNode) tomlValue).elements();
+            validateArraySize(tomlValue, variableName, arrayType, tableNodeList.size());
+            for (TomlValueNode tomlTableNode : tableNodeList) {
+                validateValue(tomlTableNode, variableName, elementType);
+            }
+            return;
+        }
+        invalidTomlLines.add(tomlValue.location().lineRange());
+        throw new ConfigException(CONFIG_INCOMPATIBLE_TYPE, getLineRange(tomlValue), variableName, arrayType,
+                getTomlTypeString(tomlValue));
+    }
+
+    private void validateArraySize(TomlNode tomlValue, String variableName, ArrayType arrayType, int tomlArraySize) {
+        int arraySize = arrayType.getSize();
```

Comment:
Add a comment to remove this check after fixing https://github.com/ballerina-platform/ballerina-lang/issues/35522